clang/test/dpct/syncthreads.cu

```diff
@@ -149,4 +161,129 @@ __global__ void test9(S4 a) {
   S4 b(a);
   // CHECK:item_ct1.barrier();
   __syncthreads();
-}
\ No newline at end of file
+}
+
+extern __shared__ float extern_local_decl[];
+
+__global__ void test10(float *arg1_ptr, int arg2_scalar, int arg3_scalar) {
+  int var1 = blockIdx.x * blockDim.x + threadIdx.x;
```

Comment:
we can check the item_ct1's type, migrate syncthread to an if-else statement.
Then we can get rid of the analysis of 1dim/3dim of kernel call.

Reply:
If we keep this 1dim/3dim analysis, we need to add more tests.
E.g.,
two different kernel calls with same global function;
no kernel call with this global function.